src/resources/prefabs.rs

```diff
@@ -0,0 +1,52 @@
+use std::collections::HashMap;
+
+use amethyst::{
+    assets::{Handle, Prefab, PrefabLoader, RonFormat},
+    ecs::World,
+};
+
+use crate::components::creatures::{CreaturePrefabData, CreatureType};
+
+#[derive(Default)]
+pub struct CreaturePrefabs {
+    prefabs: HashMap<CreatureType, Handle<Prefab<CreaturePrefabData>>>,
+}
+
+impl CreaturePrefabs {
+    pub fn insert(
+        &mut self,
+        creature_type: CreatureType,
+        prefab_handle: Handle<Prefab<CreaturePrefabData>>,
+    ) {
+        self.prefabs.insert(creature_type, prefab_handle);
+    }
+
+    pub fn get_prefab(
+        &self,
+        creature_type: &CreatureType,
+    ) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
+        self.prefabs.get(creature_type)
+    }
+}
+
+// Here we load all prefabs for the different creatures in the game.
+// These prefabs are then stored in a resource of type CreaturePrefabs that is used by the spawner system.
+pub fn initialize_prefabs(world: &mut World) {
+    let mut creature_prefabs = CreaturePrefabs::default();
+    let carnivore_prefab = world.exec(|loader: PrefabLoader<'_, CreaturePrefabData>| {
```

Comment:
Can't this be done in one `exec`?

Reply:
It can. I changed it. Now, it loads all three prefabs in one `exec`. Ultimately, I'd like to be able to list all prefabs in a directory (`prefabs/creatures`probably), load them all and store them with a `String` identifier instead of an enum.